### PR TITLE
PROD-30974: Implement "user logs in" event for the EDA

### PR DESCRIPTION
**Pull request by tregismoreira**
> [!CAUTION]
> This PR depends on #4143

## Description
This pull request introduces the new EDA event "user logs in" which is dispatches to the Kafka topic `com.getopensocial.cms.user.login`.

Here's an example of the generated payload:

```
{
	"specversion": "1.0",
	"id": "3f20068e-c6e2-4bc5-89c7-a2e91164656a",
	"source": "/user/login",
	"type": "com.getopensocial.cms.user.login",
	"datacontenttype": "application/json",
	"time": "2024-10-24T12:15:59Z",
	"data": {
		"user": {
			"id": "3cd53a66-b86a-41b4-8225-66eed5bb5fe3",
			"created": "2024-10-24T12:15:59",
			"updated": "2024-10-24T12:15:59",
			"status": "active",
			"displayName": "admin",
			"firstName": "",
			"lastName": "",
			"email": "devel@getopensocial.com",
			"roles": [
				"authenticated",
				"administrator"
			],
			"timezone": "UTC",
			"language": "en",
			"address": {
				"label": "",
				"countryCode": "",
				"administrativeArea": "",
				"locality": "",
				"dependentLocality": "",
				"postalCode": "",
				"sortingCode": "",
				"addressLine1": "",
				"addressLine2": ""
			},
			"phone": "",
			"function": "",
			"organization": "",
			"href": {
				"canonical": "http://cablecar.localhost/user/1/home"
			}
		},
		"actor": {
			"application": null,
			"user": {
				"id": "3cd53a66-b86a-41b4-8225-66eed5bb5fe3",
				"displayName": "admin",
				"href": {
					"canonical": "http://cablecar.localhost/user/1/home"
				}
			}
		}
	}
}
```

## Release notes (to customers)
N/A

## Issue tracker
https://getopensocial.atlassian.net/browse/PROD-30974

## Theme issue tracker
N/A

## How to test
- [ ] Checkout branch `issue/PROD-30974-eda-user-login`
- [ ] Enable modules `social_eda` and `social_eda_dispatcher`
- [ ] Log in as any user

The message should be dispatched to Kafka. If you are running locally using the Docker containers, you should be able to see the message on the Kafka UI at http://localhost:8080/.

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4151-dhpz1rlr7aso5gxunkkqunxzrq6pdx3c.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/672381c05b43db33793a884f

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4151-dhpz1rlr7aso5gxunkkqunxzrq6pdx3c.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/672381c05b43db33793a884f

File: modules/social_features/social_user/asyncapi.yml [hunk 1]
```diff
@@ -269,6 +269,141 @@ channels:
...
+                      nullable: true
+                      description: The first name of the user.
+                    lastName:
+                      type: string
+                      nullable: true
+                      description: The last name of the user.
+                    email:
+                      type: string
+                      format: email
+                      description: The email address of the user.
```
Comment (bramtenhove) [inline modules/social_features/social_user/asyncapi.yml:315]:
```suggestion
```

As part of the payload design we left this out.

File: modules/social_features/social_user/asyncapi.yml [hunk 2]
```diff
@@ -269,6 +269,141 @@ channels:
...
+                      nullable: true
+                      description: The phone number of the user.
+                    function:
+                      type: string
+                      nullable: true
+                      description: The job function of the user.
+                    organization:
+                      type: string
+                      nullable: true
+                      description: The organization of the user.
```
Comment (bramtenhove) [inline modules/social_features/social_user/asyncapi.yml:369]:
```suggestion
```

As part of the payload design we left this out.

**Review by bramtenhove — CHANGES_REQUESTED** [review]
I noticed the payload should contain less data, I made suggestions for that.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4151-dhpz1rlr7aso5gxunkkqunxzrq6pdx3c.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/672381c05b43db33793a884f

File: modules/social_features/social_user/asyncapi.yml (hunk 2 above)
Comment (tregismoreira) [reply modules/social_features/social_user/asyncapi.yml:369]: @bramtenhove You are right. It's fixed. Thanks!

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4151-dhpz1rlr7aso5gxunkkqunxzrq6pdx3c.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/672381c05b43db33793a884f

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4151-dhpz1rlr7aso5gxunkkqunxzrq6pdx3c.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/672381c05b43db33793a884f

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4151-dhpz1rlr7aso5gxunkkqunxzrq6pdx3c.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/672381c05b43db33793a884f

**Review by ribel — APPROVED** [review]
I have tested changes locally and it works as per payload design.

image

**Pull request merged**